pkg/openapi/validation_test.go

```diff
@@ -46,6 +46,14 @@ func Test_ValidateMutationPolicy(t *testing.T) {
 		},
+		{
+			description: "Policy with patchesJson6902 and added element at the beginning of a list",
+			policy:      []byte(`{"apiVersion": "kyverno.io/v1","kind": "ClusterPolicy","metadata": {"name": "pe"},"spec": {"rules": [{"name": "pe","match": {"resources": {"kinds": ["Endpoints"]}},"mutate": {"patchesJson6902": "- path: \"/subsets/0/addresses/0\"\n  op: add\n  value: {\"ip\":\"123\"}\n- path: \"/subsets/1/addresses/0\"\n  op: add\n  value: {\"ip\":\"123\"}"}}]}}`),
+		},
+		{
+			description: "Policy with patchesJson6902 and added element at the end of a list",
+			policy:      []byte(`{"apiVersion": "kyverno.io/v1","kind": "ClusterPolicy","metadata": {"name": "pe"},"spec": {"rules": [{"name": "pe","match": {"resources": {"kinds": ["Endpoints"]}},"mutate": {"patchesJson6902": "- path: \"/subsets/0/addresses/-\"\n  op: add\n  value: {\"ip\":\"123\"}\n- path: \"/subsets/1/addresses/-\"\n  op: add\n  value: {\"ip\":\"123\"}"}}]}}`),
```

Comment:
This test currently fails - the jsonpatch path is `/subsets/1/addresses/-`

It _seems_ like the trailing `-` is interpreted as the start of an `array` when in `jsonpatch` it can signal that an entry should be added to the end of a `map`.  So validation logic expects an `array` and receives a `map` -> error is wrongfully emitted.

Reply:
Yes you are right. Should we merge this PR after the fix for schema validation? BTW, I don't think this is the only issue we had with schema validation, so created another issue to disable it https://github.com/kyverno/kyverno/issues/2185. Do you think we should add it or fix the validation logic directly?

Reply:
I have added a fix to the `json-patch` library:
https://github.com/evanphx/json-patch/pull/148

Reply:
@MarcelMue - let me know when the PR is ready to merge.

Reply:
Yes - library fix is not merged yet. The maintainer might be busy / on vacation, so waiting a bit longer.